### ISSUE TITLE
When I run an application in Wayland, the image of the second monitor is copied to the first monitor.

### ISSUE THREAD

**Issue by imLEGR** [flameshot-org/flameshot#2970]
### Flameshot Version

Flameshot v12.1.0 (-)
Compiled with Qt 5.15.5

### Installation Type

User repository (AUR)

### Operating System type and version

Arch Linux

### Description

When I run an application in Wayland, the image of the second monitor is copied to the first monitor. Unable to take a screenshot. (Intel processor, Nvidia graphics card) 

### Steps to reproduce

1) I press Ctrl+PrScr (Flameshot runs on this key).
2) There is a bug, more details in the attached video.

### Screenshots or screen recordings

https://user-images.githubusercontent.com/53097257/200051570-968d7d02-0ead-4759-a319-ec148e98090e.mp4

### System Information

Operating System: Arch Linux
KDE Plasma Version: 5.26.2
KDE Frameworks Version: 5.99.0
Qt Version: 5.15.7
Kernel Version: 6.0.6-zen1-1-zen (64-bit)
Processors: 16 × Intel® Core™ i7-10700K CPU @ 3.80GHz
Memory: 62.7 ГиБ of RAM
Graphics Processor: NVIDIA GeForce RTX 3070 Ti/PCIe/SSE2
Manufacturer: Micro-Star International Co., Ltd.
Product Name: MS-7C80
System Version: 1.0

**Comment by tjb0607** [comment]
I get the same issue in Sway, it seems to be taking a full screenshot of the entire wayland session but then attempting to show that screenshot on just the active monitor, which doesn't have room to fit the whole thing

**Comment by mmahmoudian** [comment]
> it seems to be taking a full screenshot of the entire wayland session

This is literally how Flameshot works. It takes a full screenshot and then show it in an overlay and let the user to select a region and annotate

**Comment by tjb0607** [comment]
Yes, but when using multiple monitors, the full Wayland session is much bigger than what can fit on one monitor. The problem here is that the Flameshot window only occupies one monitor, rather than stretching across the entire session.

For me on Sway, this is all that I see of Flameshot, because the Flameshot window only shows on my primary display. I only see the top left corner of my session, and everything else is cut off. You can see in wdisplays how much else is being cut off:
![image](https://user-images.githubusercontent.com/1190870/200163881-9b59feb0-93e5-44ca-a282-9ad34c8a3576.png)

Using Sway, it's possible to work around this by taking Flameshot out of fullscreen and making it floating, then manually moving it into the right spot, but that's a pain to do every single time.

**Comment by mmahmoudian** [comment]
@tjb0607 I didn't say it is not a bug, I only made an statement/clarification/confirmation on one sentence. ;)

**Comment by mmahmoudian** [comment]
Seems related to #2364

**Comment by artur-kabirov** [comment]
Seems also related to [#2962 ](https://github.com/flameshot-org/flameshot/issues/2962)

**Comment by enzinia** [comment]
This branch looks related, still not in the master tho.
https://github.com/flameshot-org/flameshot/compare/master...fix_wayland_multiscreen

Will be great if someone can build from it and check.
(I have the same problem on Ubuntu 22.10)

**Comment by bblacher** [comment]
You can fix this in sway by adding this line to your sway config:
`for_window [app_id="flameshot"] floating enable, fullscreen disable, move absolute position 0 0, border pixel 0`

**Comment by valeth** [comment]
A similar fix for KDE
Create a new Window rule with the following properties:

`Window class = flameshot`
`Window title = flameshot` (otherwise it would affect the config window)
`Position = Force, 0 x 0`
`Fullscreen = Force, No`

**Comment by imLEGR** [comment]
> A similar fix for KDE Create a new Window rule with the following properties:
> 
> `Window class = flameshot` `Window title = flameshot` (otherwise it would affect the config window) `Position = Force, 0 x 0` `Fullscreen = Force, No`

Can you please tell me how to do what you just wrote?

**Comment by valeth** [comment]
@imLEGR you go to `System Settings > Window Management > Window Rules`
Create a new rule, name it however you like.
Then add the config like this.
![image](https://user-images.githubusercontent.com/3198362/206327175-0e2841a1-4d06-403b-a6cd-1e1c37f9d043.png)

**Comment by imLEGR** [comment]
@valeth  Thank you so much. Everything works for me now. I also had this rule, so I removed it. Apparently, the rule on the screenshot also somehow interfered with the correct operation.
![Screenshot](https://user-images.githubusercontent.com/53097257/206328665-55c9bcc9-9f58-4a5a-b613-82599f4bd779.jpg)

**Comment by AkechiShiro** [comment]
@enzinia have you tried that branch under Sway ? Does it fix the issue ?

**Comment by yajo** [comment]
This started happening to me with NixOS 25.11 on GNOME. Did anybody find a workaround for GNOME?

**Comment by Kerwood** [comment]
This is happening to me as well on multiple PCs

- Fedora 42, Gnome 48, Wayland, Flameshot v13.1.0
- Fedora Silverblue 42, Gnome 48, Wayland, Flameshot v13.1.0

As a workaround, this works.
```
QT_QPA_PLATFORM=xcb flameshot gui
```

**Comment by mmahmoudian** [comment]
This should have been fixed with #4498 
If you can, try compiling from the master branch and let us know is this issue still persists. I'm going to close this for now, but if the issue persists, we will re-open it.
